### PR TITLE
Complex geom

### PR DESCRIPTION
**Pull request by sgbalogh**
Addresses #524 –– adds ability to parse Solr ENVELOPE or WKT polygons from record into GeoJSON Leaflet layers in order to support complex display geometries.

Would be nice to have some tests focusing on the new functionality, but any fixtures with WKT polygons will require a Solr install with JTS to support. Perhaps a compromise would be just testing [these functions](https://github.com/geoblacklight/geoblacklight/blob/complex_geom/app/models/concerns/geoblacklight/solr_document/geometry.rb)?

### CODE REVIEW

File: app/assets/javascripts/geoblacklight/modules/results.js [hunk 1]
```diff
@@ -1,76 +1,80 @@
-Blacklight.onLoad(function() {
-  var historySupported = !!(window.history && window.history.pushState);
+Blacklight.onLoad(function () {
+    var historySupported = !!(window.history && window.history.pushState);
```
Comment (mejackreed) [inline app/assets/javascripts/geoblacklight/modules/results.js]: Are these just whitespace changes? If so can we move these to separate commits? 

We standardize on two spaces everywhere, it looks like this is using 4 spaces? I realize that we could probably  update the tooling to better support this. https://github.com/geoblacklight/geoblacklight/blob/master/.jshintrc#L32

**Comment by sgbalogh** [comment]
@mejackreed thanks for taking a look! I think you're right about the whitespace –– I'll look at that.

About the GeoJSON indexing –– I did experiment with that a little (though in general there is not much in the way of documentation on it). I noticed, though, that enabling it (which you have to do by adding a `format="GeoJSON"` tag to the spatial field in schema.xml) seems to force all geometries to be in GeoJSON, and (stranger) only responds to spatial queries that use GeoJSON. The ENVELOPE syntax is, I guess, just a flavor of WKT though –– so it seemed like sticking with WKT for complex polygons might be the best way not to break backward compatibility with records. But maybe there is a way around this that I just haven't seen documented.

And the feature is always active –– basically, if there a complex polygon in WKT as part of the spatial field in a record, it will now be displayable in the results page, or shown on a show page if applicable. Spatial queries are still sent as rectangles from the Leaflet viewport, but being able to index complex shapes helps immensely with spatial search for records with coverage as large as the USA, for example.

A potential risk would be in bombarding the DOM with a ton of complex GeoJSON layers on the results page, but I'm not convinced this would be a serious problem, especially since all layers are displayed one at a time.

**Comment by mejackreed** [comment]
Gotcha about the GeoJSON.

I do have concerns about adding this much functionality without having end to end feature tests. I wonder if we could figure out to add JTS into solr wrapper for our test suite?

**Comment by mejackreed** [comment]
Maybe https://github.com/cbeer/solr_wrapper/blob/master/lib/solr_wrapper/instance.rb#L30 this could help us?

**Comment by sgbalogh** [comment]
I'm going to close this, since it seems related to issues we want to talk through more.
